core/state/statedb_test.go

```diff
@@ -984,3 +988,72 @@ func makeBLSPubSigPair() blsPubSigPair {
 
 	return blsPubSigPair{shardPub, shardSig}
 }
+
+func TestValidatorRevert(t *testing.T) {
```

Comment:
Can you add more test cases? To cover more situations like modify,modify,revert; modify,modify,revert,revert. Or updating other fields rather than just the delegations. The test coverage right now is not high.

Reply:
I have added more tests, please see `TestValidatorMultipleReverts` in particular. The coverage for new code in `statedb.go` is fairly high.